code/modules/projectiles/guns/energy/experimental.dm

```diff
@@ -506,3 +506,85 @@
 			detonate()
 
 #undef MAX_STICKYBOMBS
+
+/obj/item/weapon/gun/nikita
+	name = "\improper Nikita"
+	desc = "A miniature cruise missile launcher. Using a pulsed rocket engine and sophisticated TV guidance system."
+	icon = 'icons/obj/gun_experimental.dmi'
+	icon_state = "nikita"
+	item_state = null
+	origin_tech = null
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/guns_experimental.dmi', "right_hand" = 'icons/mob/in-hand/right/guns_experimental.dmi')
+	recoil = 1
+	flags = FPRINT
+	w_class = 4.0
+	fire_delay = 2
+	fire_sound = 'sound/weapons/rocket.ogg'
+	var/loaded = 0
+	var/obj/item/projectile/nikita/fired = null
+	var/emagged = 0
+
+/obj/item/weapon/gun/nikita/examine(mob/user)
+	..()
+	if(loaded)
+		user << "<span class='info'>It appears to be loaded.</span>"
+
+/obj/item/weapon/gun/nikita/update_icon()
+	return
+
+/obj/item/weapon/gun/nikita/attack_self(mob/user)
+	if(fired)
+		playsound(get_turf(src), 'sound/weapons/stickybomb_det.ogg', 30, 1)
+		fired.detonate()
+
+/obj/item/weapon/gun/nikita/suicide_act(var/mob/user)
+	if(!loaded)
+		user.visible_message("<span class='danger'>[user] jams down \the [src]'s trigger before noticing it isn't loaded and starts bashing \his head in with it! It looks like \he's trying to commit suicide.</span>")
+		return(BRUTELOSS)
+	else
+		user.visible_message("<span class='danger'>[user] fiddles with \the [src]'s safeties and suddenly aims it at \his feet! It looks like \he's trying to commit suicide.</span>")
+		spawn(10) //RUN YOU IDIOT, RUN
+			explosion(src.loc, -1, 1, 4, 8)
+			return(BRUTELOSS)
+	return
+
+/obj/item/weapon/gun/nikita/attackby(var/obj/item/A as obj, mob/user as mob)
+	if(istype(A, /obj/item/nikita))
```

Comment:
You should be using the same system as all guns for things like this, not coding special cases.